x-pack/plugins/secops/server/lib/network_top_n_flow/elasticsearch_adapter.ts

```diff
@@ -30,37 +31,9 @@ export class ElasticsearchNetworkTopNFlowAdapter implements NetworkTopNFlowAdapt
       buildQuery(options)
     );
     const { cursor, limit } = options.pagination;
-    const totalCount = getOr(0, 'aggregations.network_top_n_flow_count.value', response);
-    const buckets = getOr([], 'aggregations.network_top_n_flow.buckets', response);
-    const networkTopNFlowItemAttr =
-      options.networkTopNFlowType === NetworkTopNFlowType.source ? 'source' : 'destination';
-    const networkTopNFlowEdges: NetworkTopNFlowEdges[] = buckets.map(
-      (bucket: NetworkTopNFlowBuckets) => ({
-        node:
-          bucket.domain.buckets.length > 0
-            ? {
-                _id: bucket.key,
-                [networkTopNFlowItemAttr]: {
-                  domain:
-                    bucket.domain.buckets[0].key === '__missing__'
-                      ? ''
-                      : bucket.domain.buckets[0].key,
-                  ip: bucket.key,
-                },
-                event: { duration: bucket.domain.buckets[0].event_duration.value },
-                network: {
-                  bytes: bucket.domain.buckets[0].network_bytes.value,
-                  packets: bucket.domain.buckets[0].network_packets.value,
-                },
-              }
-            : {},
-        cursor: {
-          value: bucket.key,
-          tiebreaker: null,
-        },
-      })
-    );
-    const hasNextPage = networkTopNFlowEdges.length === limit + 1;
+    const totalCount = getOr(0, 'aggregations.top_n_flow_count.value', response);
+    const networkTopNFlowEdges: NetworkTopNFlowEdges[] = getTopNFlowEdges(response, options);
+    const hasNextPage = networkTopNFlowEdges.length > limit;
```

Comment:
Below this line you don't need the `!` anymore since that you have a null check around it:

```ts
const beginning = cursor != null ? parseInt(cursor, 10) : 0;
```